### ISSUE TITLE
Stacktrace displayed when adding an invalid expense

### ISSUE THREAD

**Issue by manparvesh** [yoda-pa/yoda#76]
# For bug reporting
## What are the steps to reproduce this issue?
1. Run `yoda money exp`
2. Enter `specdt 30 dollars on icecream`
3. This response will be saved to the expense, but while reading this will show an error.
4. Run `yoda money exps` to show the expense

## What happens?
A stacktrace will be shown

## What were you expecting to happen?
No stacktrace should be shown.

## Implementation ideas
This invalid expense shouldn't have been added to the expense list in the first place. Just show an error while adding this expense.

## What versions of software are you using?
**Operating System:** Ubuntu 16.04

**Comment by WCollins3** [comment]
I wasn't able to replicate the issue in the exact way that you did (the word "specdt" didn't seem to break it), but I did look for other entries that could break it (invalid number and currency values). I wrote in checks that should avoid recording information that is invalid based on these two values not being correct. Since the word "specdt" does not seem to be causing the issue (or getting recorded), I did not include a check for that word. One thing that I did think was worth bringing up was that the item being purchased is only being recorded as one word, so recording a purchase of "peanut butter" may be recorded as a purchase of "butter". Not sure if this is intended, or if this is something you wanted to make another issue for.
